phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -785,19 +785,33 @@ private boolean avoidRpcToGetTable(boolean alwaysHitServer, Long resolvedTimesta
             final long effectiveUpdateCacheFreq;
             final String ucfInfoForLogging; // Only used for logging purposes
 
+            boolean overrideUcfToDefault =
```

Comment:
@virajjasani Should we override UCF if the ptable passed to this method is an index in any "unusable" state and not just PENDING_DISABLE?

Reply:
Sounds good, let me extend this support as well.

Reply:
@virajjasani thanks, this looks good now!